sdk/communication/AzureCommunication/Source/CommunicationIdentifierSerializer.swift

```diff
@@ -43,32 +43,76 @@ public class CommunicationIdentifierSerializer {
+            }
+            guard let cloud = identifier.cloud else {
+                throw AzureError.client("Can't serialize CommunicationIdentifierModel: cloud is undefined.")
+            }
+            return MicrosoftTeamsUserIdentifier(userId: microsoftTeamsUserId, isAnonymous: isAnonymous, id:id, cloudEnvironment: try deserialize(model: cloud))
         default:
             return UnknownIdentifier(identifier: id)
         }
     }
 
-    static func serialize(identifier: CommunicationIdentifier) throws -> CommunicationIdentifierModel? {
+    static func serialize(identifier: CommunicationIdentifier) throws -> CommunicationIdentifierModel {
         switch identifier {
         case let userIdentifier as CommunicationUserIdentifier:
             return CommunicationIdentifierModel(kind: .communicationUser, id: userIdentifier.identifier)
         case let callingApplicationIdentifier as CallingApplicationIdentifier:
             return CommunicationIdentifierModel(kind: .callingApplication, id: callingApplicationIdentifier.identifier)
         case let phoneNumberIdentifier as PhoneNumberIdentifier:
-            return CommunicationIdentifierModel(kind: .phoneNumber, phoneNumber: phoneNumberIdentifier.phoneNumber)
+            return CommunicationIdentifierModel(kind: .phoneNumber, id: phoneNumberIdentifier.id, phoneNumber: phoneNumberIdentifier.phoneNumber)
         case let microsoftTeamUserIdentifier as MicrosoftTeamsUserIdentifier:
             return CommunicationIdentifierModel(
                 kind: .microsoftTeamsUser,
-                id: microsoftTeamUserIdentifier.userId,
-                isAnonymous: microsoftTeamUserIdentifier.isAnonymous
+                id: microsoftTeamUserIdentifier.id,
+                microsoftTeamsUserId: microsoftTeamUserIdentifier.userId,
+                isAnonymous: microsoftTeamUserIdentifier.isAnonymous,
+                cloud: try serialize(cloud: microsoftTeamUserIdentifier.cloudEnviroment)
             )
         case let unknownIdentifier as UnknownIdentifier:
             return CommunicationIdentifierModel(kind: .unknown, id: unknownIdentifier.identifier)
         default:
             throw AzureError.client("Not support kind in CommunicationIdentifier.")
         }
     }
+
+    private static func deserialize(model: CommunicationCloudEnvironmentModel) throws -> CommunicationCloudEnvironment {
+        if (model == CommunicationCloudEnvironmentModel.Public) {
+            return CommunicationCloudEnvironment.Public;
+        }
+        if (model == CommunicationCloudEnvironmentModel.Gcch) {
+            return CommunicationCloudEnvironment.Gcch;
+        }
+        if (model == CommunicationCloudEnvironmentModel.Dod) {
+            return CommunicationCloudEnvironment.Dod;
+        }
+
+        throw AzureError.client("Not support kind in CommunicationIdentifier.")
```

Comment:
We shouldn't throw here but instead 
```suggestion
        return CommunicationCloudEnvironment(cloud.requestString);
```

Reply:
Good catch.